src/frequenz/sdk/_internal/_logging.py

```diff
@@ -0,0 +1,300 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Logging utilities for the SDK."""
+
+import logging
+from collections.abc import Mapping
+from datetime import datetime, timedelta
+from types import TracebackType
+
+_ExcInfoType = (
+    bool
+    | BaseException
+    | tuple[None, None, None]
+    | tuple[type[BaseException], BaseException, TracebackType | None]
+    | None
+)
+
+DEFAULT_RATE_LIMIT = timedelta(minutes=15)
```

Comment:
I would probably just put the literal `timedelta()` in the constructor to avoid the indirection in the docs. If you keep it this way, you should documented via a docstring so it appears in the docs and users can know what's the default.